packages/firebase_admob/lib/firebase_admob.dart

```diff
@@ -184,6 +184,84 @@ class InterstitialAd extends MobileAd {
   Future<bool> load() => _doLoad("loadInterstitialAd");
 }
 
+/// [RewardedVideoAd] status changes reported to [RewardedVideoAdListener]s.
+///
+/// The [rewarded] event is particularly important, since it indicates that the
+/// user has watched a video to completion and should be given an in-app reward.
+enum RewardedVideoAdEvent {
+  loaded,
+  failedToLoad,
+  opened,
+  leftApplication,
+  closed,
+  rewarded,
+  started,
+}
+
+/// Signature for a [RewardedVideoAd] status change callback. The optional
+/// parameters are only used when the [RewardedVideoAdEvent.rewarded] event
+/// is sent, and will be null for all others.
+typedef void RewardedVideoAdListener(RewardedVideoAdEvent event,
+    [String rewardType, int rewardAmount]);
+
+/// The AdMob rewarded video ad. The AdMob API uses a singleton for its rewarded
+/// video ads, and this class is designed to match.
+///
+/// Apps should assign a callback function to [RewardedVideoAd]'s listener
+/// property in order to receive reward notifications from the AdMob SDK:
+/// ```
+/// RewardedVideoAd.instance.listener = (RewardedVideoAdEvent event,
+///     [String rewardType, int rewardAmount]) {
+///     print("You were rewarded with $rewardAmount $rewardType!");
+///   }
+/// };
+/// ```
+///
+/// The function will be invoked when any of the events in
+/// [RewardedVideoAdEvent] occur.
+///
+/// To load and show ads, call the load method:
+/// ```
+/// RewardedVideoAd.instance.load(myAdUnitString, myTargetingInfoObj);
+/// ```
+///
+/// Later (any point after your listener callback receives the
+/// RewardedVideoAdEvent.loaded event), call the show method:
+/// ```
+/// RewardedVideoAd.instance.show();
+/// ```
+///
+/// Only one rewarded video ad can be loaded at a time. Because the creatives
+/// are so large, it's a good idea to start loading an ad well in advance of
+/// when it's likely to be needed.
+class RewardedVideoAd {
```

Comment:
This class uses `adUnitId` and the existing classes use `unitId`. I think the latter is OK, since they're all ads. We should be consistent.

Reply:
I agree that they should all be consistent, but I need to push back on the particular name "unitId." We never refer to an AdMob ad unit ID without using the entire phrase "ad unit ID." "Unit ID" on its own never shows up in our API, documentation, or developer console.

I would have renamed the ad unit ID parameters for banners and interstitials, but that's a breaking change, and I wasn't sure if it was advisable/allowed. 

What would you recommend here?

Reply:
It's very early days for this plugin, I think a breaking change is warranted.  So `adUnitId` it is, everywhere.